ta_services/secure_key_services/include/sks_ta.h

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2017, Linaro Limited
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#ifndef __SKS_TA_H__
+#define __SKS_TA_H__
+
+#include <sys/types.h>
+#include <stdint.h>
+
+#define TA_SKS_UUID { 0xfd02c9da, 0x306c, 0x48c7, \
+                        { 0xa4, 0x9c, 0xbb, 0xd8, 0x27, 0xae, 0x86, 0xee } }
+
+#define KEYCMD			0
+#define PROCCMD			1
+#define DEBUGCMD		2
+#define CRYPTOKICMD		3
+
+#define TA_SKS_CMD_REQUEST_MASK		0xFFFFFFF
+
+#define TA_SKS_CMD_KEY(cmd)		((cmd) | (KEYCMD << 28))
+#define TA_SKS_CMD_PROCESSING(cmd)	((cmd) | (PROCCMD << 28))
+#define TA_SKS_CMD_DEBUG(cmd)		((cmd) | (DEBUGCMD << 28))
+#define TA_SKS_CMD_CRYPTOKI(cmd)	((cmd) | (CRYPTOKICMD << 28))
+
+#define TA_SKS_CMD(cmd)		((cmd) | (CKCMD << 28))
+#define TA_SKS_CMD2REQ(cmd)	((cmd) & TA_SKS_CMD_REQUEST_MASK)
+
+/*
+ * PKCS#11 specific behaviour against token/session/login.
+ * Specific request from the the cryptoki API
+ *
+ * SKS_CMD_CK_PING			do nothing but invoke the SKS TA.
+ * SKS_CMD_CK_SLOTS_INFO		get slot IDs (+ info) supported by the TA.
+ * SKS_CMD_CK_GET_TOKEN_INFO		get token info (Cpryptoki)
+ */
+
+/*
+ * request args:	none
+ * input data:		none
+ * output data:		none
+ */
+#define SKS_CMD_CK_PING			TA_SKS_CMD_CRYPTOKI(0x000000)
+
+/*
+ * request args:	[32b-slot-id]
```

Comment:
For all the command ids: please specify exactly how each of the used four parameters are used.